circuit/src/execution-delivery/src/exec_composer/mod.rs

```diff
@@ -849,4 +892,303 @@ mod tests {
+
+        ext.execute_with(|| {
+            insert_default_xdns_record();
+            let res = ExecComposer::run_single_contract::<Test, OptimisticOutputMode>(
+                &mut temp_contract_one,
+                Default::default(),
+                Weight::MAX,
+                Default::default(),
+                Default::default(),
+                Default::default(),
+                Default::default(), // requester
+                Default::default(),
+                Default::default(),
+            );
+
+            assert_eq!(res, Err("Can't create VVM call stack"));
+        });
+    }
+
+    #[test]
+    fn run_single_contract_fails_xdns_error() {
+        let mut ext = TestExternalities::new_empty();
+        let requester =
+            AccountId32::from_str("5G9VdMwXvzza9pS8qE8ZHJk3CheHW9uucBn9ngW4C1gmmzpv").unwrap();
+        let value = BalanceOf::<Test>::from(0u32);
+        let mut temp_contract_one =
+            make_registry_contract_out_of_wat::<Test>(CODE_CALL, vec![], requester.clone(), value);
+
+        ext.execute_with(|| {
+            // This comment line is intentional
+            // Not inserting xdns record to replicate this failure
+            // insert_default_xdns_record();
+
+            let res = ExecComposer::run_single_contract::<Test, OptimisticOutputMode>(
+                &mut temp_contract_one,
+                Default::default(),
+                Weight::MAX,
+                Default::default(),
+                Default::default(),
+                Default::default(),
+                Default::default(),
+                Default::default(),
+                Default::default(),
+            );
+
+            assert_eq!(res, Err("Xdns record not found"));
+        });
+    }
+
+    #[test]
+    fn run_single_contract_fails_wasm_parse_error() {
+        let mut ext = TestExternalities::new_empty();
+        let requester =
+            AccountId32::from_str("5G9VdMwXvzza9pS8qE8ZHJk3CheHW9uucBn9ngW4C1gmmzpv").unwrap();
+        let value = BalanceOf::<Test>::from(0u32);
+        let mut temp_contract_one = make_registry_contract_out_of_wat::<Test>(
+            INVALID_CODE,
+            vec![],
+            requester.clone(),
+            value,
+        );
+        ext.execute_with(|| {
+            insert_default_xdns_record();
+            let res = ExecComposer::run_single_contract::<Test, OptimisticOutputMode>(
+                &mut temp_contract_one,
+                Default::default(),
+                Weight::MAX,
+                Default::default(),
+                Default::default(),
+                Default::default(),
+                Default::default(),
+                Default::default(),
+                Default::default(),
+            );
+
+            assert_eq!(res, Err("Can't decode WASM code"));
+        });
+    }
+
+    #[test]
+    #[ignore = "Please implement proper assertions when retrieve_gateway_protocol is properly implemented"]
```

Comment:
Yeah this needs #87 to be merged first.

Reply:
I will let this test stay.
And I have refactored the tests out into a new file.
I am resolving this comment.